DB設計

gingin-s opened the pull request:
# What
 ER図の作成
READMEの編集

# Why
DB設計のため


ER図(Gyazo URL)
https://i.gyazo.com/6c31ab1ab6ec9b3b6df69d787e3a4bdc.png

gingin-s committed:
ER図の作成

gingin-s committed:
READMEの編集

CHURO-MAU commented on README.md:
```diff
@@ -1,24 +1,79 @@
+| last_name             | string         | null: false                 |
+| first_name_kana       | string         | null: false                 |
+| last_name_kana        | string         | null: false                 |
+| birth_year_id         | integer        | null: false                 |
+| birth_month_id        | integer        | null: false                 |
+| birth_dates_id        | integer        | null: false                 |
```
１４〜１６行目
生年月日は一つのカラムにまとめることが可能です。
Typeはdate型を使用し、一つのカラムで設計してみましょう。

補足
今回のアプリにでは、生年月日はActiveHashを利用しておりません。

CHURO-MAU commented on README.md:
```diff
@@ -1,24 +1,79 @@
-* Configuration
+- has_many :items
+- has_many :orders
+- belongs_to :birth_year
+- belongs_to :birth_month
+- belongs_to :birth_date
```
２３〜２５行目
こちらのアソシエーションは不要となりますので、削除しておきましょう。

CHURO-MAU commented on README.md:
```diff
@@ -1,24 +1,79 @@
+|-----------------------|----------------|--------------------------------|
+| title                 | string         | null: false                    |
+| description           | text           | null: false                    |
+| price                 | text           | null: false                    |
```
３３行目
価格のカラムの型は、整数型に修正しておきましょう。

CHURO-MAU commented on README.md:
```diff
@@ -1,24 +1,79 @@
-* Services (job queues, cache servers, search engines, etc.)
+- has_one :order
+- belongs_to :user
+- belongs_to :category_id
+- belongs_to :condition_id
+- belongs_to :postage_id
+- belongs_to :prefecture_id
+- belongs_to :shipping_date_id
```
４５〜４９行目
データベース設計ではテーブルの関連付けを記載いただければと思います。
そのため、ActiveHashのファイルとの関連付けは削除しておきましょう。

CHURO-MAU commented on README.md:
```diff
@@ -1,24 +1,79 @@
+| post_code             | string         | null: false                    |
+| city                  | string         | null: false                    |
+| address               | string         | null: false                    |
+| building_name         | string         | null: false                    |
```
７１行目
建物名は入力任意項目となりますので、NOT NULL制約は外しておきましょう。

CHURO-MAU commented on README.md:
```diff
@@ -1,24 +1,79 @@
+### Association
+
+- belongs_to :order
+- belongs_to :prefecture_id
```
７９行目
こちらもActiveHashのファイルとの記載となりますので、削除しておきましょう。

CHURO-MAU reviewed (commented):
レビュー依頼ありがとうございます！
コメントいたしましたので、ご確認お願いいたします。

次回レビュー依頼時には、最新のER図もご提出の程よろしくお願いいたします。

gingin-s committed:
REDME修正

ActiveHashに関する記述の削除
usersテーブルbirthdayカラム修正
ship_addressesデーブルbuilding_nameのoption修正

gingin-s committed:
ER図の修正

birth_years・birth_months・birth_datesの削除

gingin-s commented:
ご指摘頂いた箇所の修正を行いました。
ER図(Gazo URL)
https://i.gyazo.com/89a8f4abc592cbee65fe13537d9d861b.png

ER図内のActiveHashに関する表示も削除したほうがよろしいでしょうか？
ご確認お願いいたします。

greenichi commented on README.md:
```diff
@@ -1,24 +1,69 @@
+| last_name             | string         | null: false                 |
+| first_name_kana       | string         | null: false                 |
+| last_name_kana        | string         | null: false                 |
+| birthday              | data           | null: false                 |
```
```
| birthday | data | null: false |
```
カラムの型は**date**型にしておきましょう。

gingin-s commented:
画像にカーソルが写り込んでしまっていたため、取り直しました。
こちらをご確認ください
ER図(Gazo URL)
https://i.gyazo.com/d65da7fb95cc3dc228698de9b4b27525.png

gingin-s commented:
画像にカーソルが写り込んでしまっていたため、撮り直しました。
こちらをご確認ください
ER図(Gazo URL)
https://i.gyazo.com/d65da7fb95cc3dc228698de9b4b27525.png

greenichi reviewed (commented):
レビュー依頼ありがとうございます。
修正点確認させていただきました。きれいなREADMEになってきましたね！
1点修正をお願いしております。ご確認いただけますと幸いです。

また、ご質問いただいたER図におけるActiveHashの記述に関して
ご質問いただきありがとうございます。
今回ActiveHash専用のテーブルを生成することはないため、削除していただければと存じます。
（ER図におけるActiveHashの記述に関してはitemsテーブル内のカラム名のみ残していただきたいです。）

修正後、コミットプッシュを確認いただき、再度**修正後のER図**を添付の上、**レビュー依頼フォーム**のご提出をいたします。
よろしくお願いいたします。

ご不明点や疑問点ございましたら、通話メンターにご相談ください！
どうぞ、よろしくお願いいたします。

gingin-s committed:
ER図の修正

ActiveHashに関する記載の削除

gingin-s committed:
READMEの修正

usersテーブルbirthdayカラムの型の変更

gingin-s commented:
ご指摘頂いた箇所の修正を行いました。
ER図(Gazo URL)
https://i.gyazo.com/04c802fc0344ae7d4af690da26454452.png

ご確認お願いいたします。

greenichi reviewed (commented):
レビュー依頼ありがとうございます。
修正点確認させていただきました。
READMEはしっかり記述できていらっしゃいます！

ER図につきまして、再度修正点ご確認いただきたい箇所がございます。
```
・usersテーブルのパスワードカラムをREADMEに合わせて変更
・外部キーカラムに関しては「_id」を削除
（reference型の場合自動で_idが付与されますが、ER図の設計の段階では
READMEに合わせて_idがない状態で記述しておくことが望ましいためです。）
```
上記ご確認をお願いいたします。

修正後、コミットプッシュを確認いただき、再度**修正後のER図**を添付の上、**レビュー依頼フォーム**のご提出をいたします。
よろしくお願いいたします。

ご不明点や疑問点ございましたら、通話メンターにご相談ください！
どうぞ、よろしくお願いいたします。

gingin-s committed:
ER図の修正

usersテーブルpasswordを、encrypted_password へ変更。
外部キーの_idを削除

gingin-s commented:
レビューありがとうございます。

ご指摘頂いた箇所の修正を行いました。
ER図(Gazo URL)
https://i.gyazo.com/b6851077cfb0e705fd6b6782cb1c9cf4.png
ご確認お願いいたします。

greenichi reviewed (commented):
修正とレビュー依頼ありがとうございます。

修正点確認いたしました。
実装できていることが確認できましたので、**LGTM**とさせていただきます。
実装お疲れ様でした。

こちらのブランチは忘れずにMergeを行っていただき、ローカルのmasterブランチにPullしましょう。
その後、新しいブランチを作成し次の実装に取り組んでいただけますと幸いです。
ここまでの実装、本当にお疲れ様でした。
引き続き、応援しております。
